Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -890,22 +893,63 @@ void MatrixWorkspace::setYUnit(const std::string &newUnit) {
   m_YUnit = newUnit;
 }
 
+/** Splits a string on the word "per" and replaces it with a Latex equivalent.
+ * E.g. "Counts per meV per hour" becomes "Counts (meV hour)$^{-1}$"
+ * If no "per"s are present leave the string as it is.
+ * @param yLabel :: The y-axis label
+ * @return std::string
+ */
+std::string replacePerWithLatex(std::string yLabel) {
+  std::vector<std::string> splitVec;
+  boost::split_regex(splitVec, yLabel, boost::regex(" per "));
+  if (splitVec.size() > 1) {
+    yLabel = splitVec[0];
+    splitVec.erase(splitVec.begin());
+    std::string unitString = boost::algorithm::join(splitVec, " ");
+    yLabel += " (" + unitString + ")$^{-1}$";
+  }
+  return yLabel;
+}
+
+/** Append the x-unit of the workspace to the y-unit label as a denominator
+ * E.g. if a workspace has y-unit label "Counts" and x-unit angstrom, the y-unit
+ * label becomes "Counts per angstrom". Or if useLatex is true "Counts per
+ * $\AA$"
+ * @param yLabel :: The y-axis label
+ * @param workspace :: Pointer to workspace
+ * @param useLatex :: Boolean, if true use latex else use ascii
+ */
+std::string appendUnitDenominatorUsingPer(std::string yLabel,
+                                          const MatrixWorkspace *workspace,
+                                          bool useLatex) {
+  if (useLatex) {
+    yLabel += " per $" + workspace->getAxis(0)->unit()->label().latex() + "$";
+  } else {
+    yLabel += " per " + workspace->getAxis(0)->unit()->label().ascii();
+  }
+  return yLabel;
+}
+
 /// Returns a caption for the units of the data in the workspace
-std::string MatrixWorkspace::YUnitLabel() const {
+std::string
+MatrixWorkspace::YUnitLabel(bool useLatex /* = false */,
+                            bool plotAsDistribution /* = false */) const {
   std::string retVal;
-  if (!m_YUnitLabel.empty())
+  if (!m_YUnitLabel.empty()) {
     retVal = m_YUnitLabel;
-  else {
+    // If the workspace is marked as a distribution its units are already
+    // correct, if the workspace is not a distribution but is being plotted as
+    // one we must adjust the unit
+    if (plotAsDistribution && !this->isDistribution())
+      retVal = appendUnitDenominatorUsingPer(retVal, this, useLatex);
+  } else {
     retVal = m_YUnit;
-    // If this workspace a distribution & has at least one axis & this axis has
-    // its unit set
-    // then append that unit to the string to be returned
-    if (!retVal.empty() && this->isDistribution() && this->axes() &&
-        this->getAxis(0)->unit()) {
-      retVal = retVal + " per " + this->getAxis(0)->unit()->label().ascii();
+    if (plotAsDistribution && !this->isDistribution()) {
```

Comment:
I think I'm misunderstanding the logic here. If the workspace is a distribution then this won't fix the unit up? Doesn't that miss some cases we have? I think perhaps the interpretation of `plotAsDistribution = false` could be confusing? Should that mean force distribution off or should it mean "use the internal flag"?

Reply:
I think I actually made a bit of a mistake here: [line 127](https://github.com/mantidproject/mantid/pull/25556/files#diff-f169ee4c92b65dcf63939bdae32f943cR127) in `PlotAxis::titleFromYData` should read 
```c++
const std::string customLabel = workspace.YUnitLabel(false, plottingDistribution)
```
So `plotAsDistribution` comes from the `plottingDistribution` variable. The docstring in `titleFromYData` for `plottingDistribution` is:
> @ param plottingDistribution If true, the Y axis has been divided by the bin width

So this could be true if the workspace is a distribution or not. If the workspace *is* a distribution then we don't want to append the x-unit or we'd end up with something like `Counts per meV per meV`. 

Perhaps I should rename `plotAsDistribution` and add a param list to the docstring?